charts/opentelemetry-collector/templates/clusterrole.yaml

```diff
@@ -9,5 +9,66 @@ metadata:
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
-rules: {{ toYaml .Values.clusterRole.rules | nindent 2 -}}
+rules:
+  {{- if .Values.clusterRole.create -}}
+  {{ toYaml .Values.clusterRole.rules | nindent 2 -}}
+  {{- end }}
+  {{- if .Values.presets.eventsCollection.enabled}}
```

Comment:
It is possible that both `.Values.clusterRole.create` and `.Values.presets.eventsCollection.enabled` are `true`, which will result in both `.Values.clusterRole.rules` and these default k8s events rules being used.  that could result in some duplication.

I think `.Values.clusterRole.rules` should take precedence so I think we can use and `else if` here.